polaris/ocean/tasks/cosine_bell/analysis.py

```diff
@@ -47,143 +34,82 @@ def __init__(self, component, resolutions, icosahedral, subdir,
-                                                'convergence_eval_time')
+        vel_pd = config.getfloat('cosine_bell', 'vel_pd')
 
         ds_mesh = xr.open_dataset(f'{mesh_name}_mesh.nc')
+        sphere_radius = ds_mesh.sphere_radius
+
         ds_init = xr.open_dataset(f'{mesh_name}_init.nc')
-        # find time since the beginning of run
-        ds = xr.open_dataset(f'{mesh_name}_output.nc')
-        for j in range(len(ds.xtime)):
-            tt = str(ds.xtime[j].values)
-            tt.rfind('_')
-            DY = float(tt[10:12]) - 1
-            if DY == convergence_eval_time:
-                sliceTime = j
-                break
-        HR = float(tt[13:15])
-        MN = float(tt[16:18])
-        t = 86400.0 * DY + HR * 3600. + MN
-        # find new location of blob center
-        # center is based on equatorial velocity
-        R = ds_mesh.sphere_radius
-        distTrav = 2.0 * 3.14159265 * R / (86400.0 * convergence_eval_time) * t
-        # distance in radians is
-        distRad = distTrav / R
-        newLon = lonCent + distRad
-        if newLon > 2.0 * np.pi:
-            newLon -= 2.0 * np.pi
-
-        # construct analytic tracer
-        tracer = np.zeros_like(ds_init.tracer1[0, :, 0].values)
-        latC = ds_mesh.latCell.values
-        lonC = ds_mesh.lonCell.values
-        temp = R * np.arccos(np.sin(latCent) * np.sin(latC) +
-                             np.cos(latCent) * np.cos(latC) * np.cos(
-            lonC - newLon))
-        mask = temp < radius
-        tracer[mask] = (psi0 / 2.0 *
-                        (1.0 + np.cos(3.1415926 * temp[mask] / radius)))
-
-        # oad forward mode data
-        tracerF = ds.tracer1[sliceTime, :, 0].values
-        rmseValue = np.sqrt(np.mean((tracerF - tracer)**2))
-
-        return rmseValue, ds_init.dims['nCells']
+        latCell = ds_init.latCell.values
+        lonCell = ds_init.lonCell.values
+
+        # distance that the cosine bell center traveled in radians
+        # based on equatorial velocity
+        distance = 2.0 * np.pi * time / (86400.0 * vel_pd)
+
+        # new location of blob center
+        lon_new = lon_center + distance
+        if lon_new > 2.0 * np.pi:
+            lon_new -= 2.0 * np.pi
+
+        x_center, y_center, z_center = lon_lat_to_cartesian(
+            lon_new, lat_center, sphere_radius, degrees=False)
+        x_cells, y_cells, z_cells = lon_lat_to_cartesian(
+            lonCell, latCell, sphere_radius, degrees=False)
+        xyz_center = Vector(x_center, y_center, z_center)
+        xyz_cells = Vector(x_cells, y_cells, z_cells)
+        ang_dist_from_center = xyz_cells.angular_distance(xyz_center)
+        distance_from_center = ang_dist_from_center * sphere_radius
+
+        bell_value = cosine_bell(psi0, distance_from_center, radius)
+        tracer1 = np.where(distance_from_center < radius,
+                           bell_value,
+                           0.0)
+        return tracer1
```

Comment:
```suggestion
        return xr.DataArray(data=tracer1, dims=('nCells',))
```
Alternatively, `latCell` and `lonCell` could be left as `DataArray`s and `np.where` could be `xr.where` but I don't think that's important and it might be unnecessarily slow.